torch_xla/csrc/convolution.cpp

```diff
@@ -224,7 +218,7 @@ xla::XlaOp BuildThnnConv2dBackwardWeight(
   const auto padding_config =
       XlaHelpers::MakeXlaPaddingConfig(XlaHelpers::I64List(padding_attr));
 
-  auto builder = grad.builder();
+  auto builder = grad_output.builder();
```

Comment:
We are a bit all over with auto and non-auto. I kinda like types, unless they take two screens to show/type.
Maybe we can have a rule that we use auto for templated types and for for-iteration-variables.

Reply:
We'll do this later.